crates/chia-protocol/src/coin_spend.rs

```diff
@@ -2,10 +2,29 @@ use chia_streamable_macro::streamable;
 
 use crate::coin::Coin;
 use crate::program::Program;
+#[cfg(feature = "py-bindings")]
+use pyo3::prelude::*;
+#[cfg(feature = "py-bindings")]
+use pyo3::types::PyType;
 
 #[streamable]
 pub struct CoinSpend {
     coin: Coin,
     puzzle_reveal: Program,
     solution: Program,
 }
+
+#[cfg(feature = "py-bindings")]
+#[pymethods]
+impl CoinSpend {
+    #[classmethod]
+    #[pyo3(name = "from_parent")]
+    pub fn from_parent(cls: &Bound<'_, PyType>, cs: Self) -> PyResult<PyObject> {
```

Comment:
actually, this *does* make me a bit nervous about performance impact. parsing coins is in the critical path in some places

Reply:
Added a check if `cls` is different from constructed Rust object and now this is only called if that's true, meaning it is skipped every time in practice.